### PR TITLE
FreeBSD compatability

### PR DESCRIPTION
**Pull request by dennisdegreef**
This is a second shot at this. Sorry for the previous inactivity.
This is related to #152

### CODE REVIEW

**Comment by dennisdegreef** [comment]
I'm now trying to write my first specfiles so tests against FreeBSD can be added :)

**Comment by luxflux** [comment]
Cool! Looking forward to them! Just ask if you need any help!

**Comment by dennisdegreef** [comment]
I've never written a specfile before, and am not that familiar with Ruby, but I think this should cover the basics?

**Comment by dennisdegreef** [comment]
Rebased because of a merge conflict

File: manifests/config.pp [hunk 1]
```diff
@@ -34,7 +34,7 @@
   if $::osfamily == 'Debian' {
     concat { '/etc/default/openvpn':
       owner => root,
-      group => root,
+      group => $root_group,
```
Comment (luxflux) [inline manifests/config.pp]: Shouldn't this be `$openvpn::params::root_group`?

File: manifests/params.pp [hunk 2]
```diff
@@ -34,9 +37,12 @@
...
     'Debian': { # Debian/Ubuntu
-      $group = 'nogroup'
-      $link_openssl_cnf = true
-      $pam_module_path = '/usr/lib/openvpn/openvpn-auth-pam.so'
+      $etc_directory     = '/etc'
+      $root_group        = 'root'
+      $group             = 'nogroup'
+      $link_openssl_cnf  = true
+      $pam_module_path   = '/usr/lib/openvpn/openvpn-auth-pam.so'
+      $load_main_service = true
```
Comment (luxflux) [inline manifests/params.pp]: What is this variable used for? I cannot find any reference to it? If it's needed, we have to add it to `Archlinux` and `RedHat` as well, don't we?

File: manifests/server.pp [hunk 3]
```diff
@@ -447,8 +447,12 @@
     if $::openvpn::params::systemd {
       $lnotify = Service["openvpn@${name}"]
     } else {
```
Comment (luxflux) [inline manifests/server.pp]: Can't we use `elsif` here? I think it becomes more readable:

```puppet
if $::openvpn::params::systemd {
  $lnotify = Service["openvpn@${name}"]

} elsif $::openvpn::params::namespecific_rclink {
  $lnotify = Service["openvpn_${name}"]

} else {
  $lnotify = Service['openvpn']
  Openvpn::Server[$name] -> Service['openvpn']

}
```

File: manifests/server.pp [hunk 4]
```diff
@@ -574,9 +580,28 @@
...
         enable   => true,
         provider => 'systemd',
-        require  => [ File["/etc/openvpn/${name}.conf"], Openvpn::Ca[$ca_name] ]
+        require  => [ File["${etc_directory}/openvpn/${name}.conf"], Openvpn::Ca[$ca_name] ]
       }
     }
   }
 
+  if $::openvpn::params::namespecific_rclink {
+    file { "/usr/local/etc/rc.d/openvpn_${name}":
```
Comment (luxflux) [inline manifests/server.pp]: Please use `$etc_directory` here as well.

File: manifests/server.pp [hunk 5]
```diff
@@ -574,9 +580,28 @@
         ensure   => running,
         enable   => true,
         provider => 'systemd',
-        require  => [ File["/etc/openvpn/${name}.conf"], Openvpn::Ca[$ca_name] ]
+        require  => [ File["${etc_directory}/openvpn/${name}.conf"], Openvpn::Ca[$ca_name] ]
       }
     }
   }
 
+  if $::openvpn::params::namespecific_rclink {
```
Comment (luxflux) [inline manifests/server.pp]: This clause does not take in account if `$::openvpn::manage_service` is `false`. At least the `service` part should be excluded.

**Comment by luxflux** [comment]
Thank you for your work! It is already in a good shape! It would be nice if you could address my comments, so I can merge it afterwards.

File: manifests/params.pp (hunk 2 above)
Comment (dennisdegreef) [inline manifests/params.pp]: Due to the way things were written previously, it came in hand for disabling the 'main' Service, it's an inverted boolean option for `namespecific_rclink`, which will have a service per server.
I removed it in `f225835`

File: manifests/server.pp (hunk 3 above)
Comment (dennisdegreef) [inline manifests/server.pp]: Fixed in `58f95da` while adding a santity check for using systemd and rclinks

File: manifests/config.pp (hunk 1 above)
Comment (dennisdegreef) [inline manifests/config.pp]: Fixed in `09cc8918`

File: manifests/server.pp (hunk 5 above)
Comment (dennisdegreef) [inline manifests/server.pp]: Fixed in https://github.com/dennisdegreef/puppet-openvpn/commit/ed100549bb4fac3a10fff22984f9396a0bce8682

**Comment by dennisdegreef** [comment]
I've addressed the feedback and opened a fresh rebased pull request in #180

**Pull request closed without merging**